metricbeat/module/prometheus/remote_write/data.go

```diff
@@ -86,18 +82,18 @@ func (p *remoteWriteEventGenerator) GenerateEvents(metrics model.Samples) map[st
 
 		data := mapstr.M{name: val}
 		e.ModuleFields["metrics"].(mapstr.M).Update(data)
+	}
 
-		if p.metricsCount {
+	if p.metricsCount {
+		for _, e := range eventList {
 			// In x-pack prometheus module, the metrics are nested under the "prometheus" key directly.
 			// whereas in non-x-pack prometheus module, the metrics are nested under the "prometheus.metrics" key.
 			// Also, it is important that we do not just increment by 1 for each e.ModuleFields["metrics"] may have more than 1 metric.
```

Comment:
This is to accomodate the histogram/summary scenario?

Reply:
Yes!